test/test_torch.py

```diff
@@ -2309,6 +2309,24 @@ def test_cat_scalars(self):
                                      'zero-dimensional.*cannot be concatenated'):
             torch.cat([x, y])
 
+    def test_cat_empty(self):
+        # FIXME: this is legacy behavior and should be removed
+        # when we support empty tensors with arbitrary sizes
+        x = torch.randn(4, 3, 32, 32)
+        empty = torch.randn(0)
+
+        res1 = torch.cat([x, empty], dim=1)
+        res2 = torch.cat([empty, x], dim=1)
+        self.assertEqual(res1, res2)
+
+        conv = torch.nn.Conv2d(3, 3, kernel_size=1)
+        res1 = torch.cat([conv(x), empty], dim=1)
+        res2 = torch.cat([empty, conv(x)], dim=1)
+        self.assertEqual(res1, res2)
+
+        res1 = torch.cat([empty, empty], dim=1)
+        self.assertEqual(res1, empty)
```

Comment:
does torch.cat([], dim=...) work?  It's a case in the wrap dim function.

Reply:
`torch.cat([], dim=...)` throws a RuntimeError on this PR and on master with the same error message. This matches numpy's behavior. I'll add an additional test case for it, thanks for the catch
![image](https://user-images.githubusercontent.com/5652049/37718964-4eae04de-2cfa-11e8-8dca-267be41e591e.png)